Print methods for pibble

philip-khor opened the pull request:
Addresses #13. 

Opening this PR just to keep track of work. I'll ping when it's ready. 

To-do: 

- [x] remove square brackets around dimensions, 
- [x] add markers for unique observations after .i, .t, .d, 
- [x] figure out how to make sure grouped tbl_pbs print information, 
- [x] write tests

philip-khor committed:
Initial commit

First pass at implementing print methods for pibble.

To-do: (1) remove square brackets around dimensions, (2) add markers for unique observations after .i, .t, .d, (3) figure out how to make sure grouped tbl_pbs print information, (4) write tests

philip-khor committed:
Merge branch 'master' into print

philip-khor committed:
import dim_desc from pillar instead of dplyr

NickCH-K commented:
As a note, you may have already noticed but I removed the `grouped_pb` type since it doesn't really need to be treated any differently from `grouped_df`. A grouped `pibble` is an object with the types `tbl_pb` and `grouped_df`.

philip-khor committed:
Merge branch 'master' into print

philip-khor committed:
add group print method, unname .i and .t attributes

note: consider changing .i attribute to similar format as groups attribute for easy computation of unique # groups. bug in distinct method also fixed - .keep_all argument treated by rlang as bare column name.

philip-khor committed:
use_tidy_style, rebuild site

philip-khor committed:
introduce basic tests for pibble, remove vec_restore tests

vec_restore tests removed because it's an internal function and we can't test it

philip-khor committed:
clarify comments

philip-khor committed:
Update test-pibble.R

package imports not needed

philip-khor commented:
@NickCH-K this PR is ready for review

NickCH-K commented:
This looks great!
